File: Python Codes/8_Reverse_String.py
class Solution:
    def reverseString(self, s):
        left = 0
        right = len(s) - 1

        while left < right:
            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1

if __name__ == "__main__":
    solution = Solution()
    s = ['S', 'A', 'K', 'S', 'H', 'I']
    solution.reverseString(s)
    
    print(''.join(s))

File: CPP Codes/8_Reverse_String.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution 
{
public:
    void reverseString(vector<char>& s) 
    {
        int left = 0;
        int right = s.size() - 1;
        
        while (left < right) 
        {
            swap(s[left], s[right]);
            left++;
            right--;
        }
    }
};

int main() 
{
    Solution solution;
    vector<char> str = {'h', 'e', 'l', 'l', 'o'};
    solution.reverseString(str);
    
    for (char c : str) 
    {
        cout << c;
    }
    cout << endl;
    
    return 0;
}
